// @loaders.gl/loader-utils/dist/es5/lib/node/fs.browser.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.writeFileSync = exports.writeFile = exports.stat = exports.readdir = exports.readFileSync = exports.readFile = exports.read = exports.open = exports.isSupported = exports.fstat = exports.createWriteStream = exports.close = exports._readToArrayBuffer = void 0;
var readdir = null;
exports.readdir = readdir;
var stat = null;
exports.stat = stat;
var readFile = null;
exports.readFile = readFile;
var readFileSync = null;
exports.readFileSync = readFileSync;
var writeFile = null;
exports.writeFile = writeFile;
var writeFileSync = null;
exports.writeFileSync = writeFileSync;
var open = null;
exports.open = open;
var close = null;
exports.close = close;
var read = null;
exports.read = read;
var fstat = null;
exports.fstat = fstat;
var createWriteStream = null;
exports.createWriteStream = createWriteStream;
var _readToArrayBuffer = null;
exports._readToArrayBuffer = _readToArrayBuffer;
var isSupported = false;
exports.isSupported = isSupported;
